### PR TITLE
(PDB-1515) Use honeysql for population queries

### PR DESCRIPTION
**Pull request by mullr**

### CODE REVIEW

File: src/puppetlabs/puppetdb/query/population.clj [hunk 1]
```diff
@@ -2,29 +2,52 @@
...
+;;; Query library
+
+(def ^:private from-all-resources-and-their-nodes
+  (hcore/build :from [:catalogs :catalog_resources :certnames]
+               :merge-where [:and
+                             [:= :catalogs.id :catalog_resources.catalog_id]
+                             [:= :certnames.certname :catalogs.certname]]))
+
+(defn- where-nodes-are-active [q]
+  (hh/merge-where q [:and
```
Comment (ajroetker) [inline src/puppetlabs/puppetdb/query/population.clj]: I believe you don't need the `and` here, that's the default conjunction.

File: src/puppetlabs/puppetdb/query/population.clj [hunk 2]
```diff
@@ -2,29 +2,52 @@
...
+                     [:= :certnames.deactivated nil]
+                     [:= :certnames.expired nil]]))
+
+(defn- select-count [q]
+  (-> q
+      (hh/select [:%count.* :c])
+      hcore/format
+      query-to-vec
```
Comment (ajroetker) [inline src/puppetlabs/puppetdb/query/population.clj]: The `(comp hcore/format query-to-vec)` might make a good helper in `puppetlabs.puppetdb.honeysql`

**Comment by ajroetker** [comment]
Where do we stand on this guy?

**Comment by mullr** [comment]
I think we agreed to merge it?

**Comment by wkalt** [comment]
I'm :-1: for reasons discussed. Don't care enough to fight about it though.

**Pull request merged**